[12.0][MIG] sale_global_discount : Migration to 12.0

manuelcalerosolis opened the pull request:
This is the migration of module : sale_global_discount to 12.0

chienandalu committed:
[ADD] sale_global_discount: New module

oca-travis committed:
[UPD] Update sale_global_discount.pot

OCA-git-bot committed:
sale_global_discount 11.0.1.0.1

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-11.0/sale-workflow-11.0-sale_global_discount
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-11-0/sale-workflow-11-0-sale_global_discount/

chienandalu committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (6 of 6 strings)

Translation: sale-workflow-11.0/sale-workflow-11.0-sale_global_discount
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-11-0/sale-workflow-11-0-sale_global_discount/es/

manuelcalerosolis committed:
[MIG] : Migration to 12.0

HaraldPanten commented:
Hi @manuelcalerosolis could you check reds?

manuelcalerosolis commented:
@HaraldPanten the travis error are like this

************* Module sales_team_security
sales_team_security/readme/CONTRIBUTORS.rst:1: [W7908(missing-newline-extrafiles), ]  Missing newline
************* Module sale_exception.models.sale_order_line
sale_exception/models/sale_order_line.py:1: [C8202(unnecessary-utf8-coding-comment), ] UTF-8 coding is not necessary
************* Module portal_sale_personal_data_only.__manifest__

They are errors of other modules, so I do not understand anything. Can you help me know what's going on?

HaraldPanten commented:
Hi @manuelcalerosolis I don't really have any idea. @ValentinVinagre could you have a look? Thx

gurneyalex commented on .vscode/settings.json:
```diff
@@ -0,0 +1,3 @@
+{
+    "restructuredtext.confPath": ""
+}
```
can you remove this file?

gurneyalex commented on sale_global_discount/models/sale_order.py:
```diff
@@ -0,0 +1,140 @@
+        if self.partner_id.customer_global_discount_ids:
+            self.global_discount_ids = (
+                self.partner_id.customer_global_discount_ids or
+                self.parnter_id.commercial_partner_id
```
```suggestion
                self.partner_id.commercial_partner_id
```

Can you check if this bug is also in the original version?

manuelcalerosolis replied:
Yes, I have suggested the module author to correct it.

gurneyalex commented on sale_global_discount/models/sale_order.py:
```diff
@@ -0,0 +1,140 @@
+    )
+
+    @api.model
+    def get_discounted_global(self, price=0, discounts=None):
+        """Compute discounts successively"""
+        discounts = discounts or []
+        if not discounts:
+            return price
+        discount = discounts.pop(0)
+        price *= 1 - (discount / 100)
+        return self.get_discounted_global(price, discounts)
```
using a recursive method here is  inefficient, and it also forces you to make copies of dicounts in other places. 

```suggestion
    def get_discounted_global(self, price=0, discounts=None):
        """Compute discounts successively
        discounts is a list of discount rates, or None
        """
        if discounts is None:
            discounts = []
        for discount in discounts:
            price *= 1 - (discount / 100)
        return price
```

gurneyalex commented on sale_global_discount/models/sale_order.py:
```diff
@@ -0,0 +1,140 @@
+            amount_discounted_untaxed = amount_discounted_tax = 0
+            for line in order.order_line:
+                discounted_subtotal = self.get_discounted_global(
+                    line.price_subtotal, discounts.copy())
```
```suggestion
                    line.price_subtotal, discounts)
```

No copy required with the implementation suggested above

gurneyalex commented on sale_global_discount/models/sale_order.py:
```diff
@@ -0,0 +1,140 @@
+        res = []
+        for tax in tax_groups:
+            tax_amount = round_curr(
+                self.get_discounted_global(tax[1], discounts.copy()))
```
```suggestion
                self.get_discounted_global(tax[1], discounts))
```

gurneyalex commented on sale_global_discount/models/sale_order.py:
```diff
@@ -0,0 +1,140 @@
+            tax_amount = round_curr(
+                self.get_discounted_global(tax[1], discounts.copy()))
+            tax_base = round_curr(
+                self.get_discounted_global(tax[2], discounts.copy()))
```
```suggestion
                self.get_discounted_global(tax[2], discounts))
```

manuelcalerosolis committed:
Delete settings.json

manuelcalerosolis committed:
Update sale_order.py

pedrobaeza committed:
[IMP] sales_team_security: Contacts permissions revamp

Restrict contacts by code, removing the need of manually manipulating the existing
record rules, that can also have some side effects.

rousseldenis committed:
[12.0][FIX] sale_order_generator: Fix view

OCA-git-bot committed:
Merge PR #1172 into 12.0

Signed-off-by rousseldenis

OCA-git-bot committed:
sale_generator 12.0.1.0.1

OCA-git-bot committed:
Merge PR #1171 into 12.0

Signed-off-by pedrobaeza

oca-travis committed:
[UPD] Update sales_team_security.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-12.0/sale-workflow-12.0-sales_team_security
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sales_team_security/

rousseldenis commented:
@manuelcalerosolis could you edit PR comment and say it depends on https://github.com/OCA/account-invoicing/pull/719 ?

rousseldenis commented:
@manuelcalerosolis For travis errors, see lint line and flake8 specific errors

quentinDupont committed:
Translated using Weblate (French)

Currently translated at 68.6% (24 of 35 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_exception
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_exception/fr/

pedrobaeza commented:
@manuelcalerosolis are you going to fix this PR?

jprokop1 committed:
Translated using Weblate (Slovak)

Currently translated at 96.8% (30 of 31 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_order_type
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_order_type/sk_SK/

manuelcalerosolis commented:
Yes of course, as soon as possible.

jprokop1 committed:
Translated using Weblate (Slovak)

Currently translated at 96.8% (30 of 31 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_order_type
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_order_type/sk_SK/

jprokop1 committed:
Translated using Weblate (Slovak)

Currently translated at 100.0% (31 of 31 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_order_type
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_order_type/sk_SK/

jprokop1 committed:
Translated using Weblate (Slovak)

Currently translated at 100.0% (31 of 31 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_order_type
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_order_type/sk/

Commit:
Merge commit 'refs/pull/1156/head' of https://github.com/OCA/sale-workflow into 12.0

Commit:
Merge commit 'refs/pull/1156/head' of https://github.com/OCA/sale-workflow into 12.0

pedrobaeza commented:
Check the fixes made in #1181 (and its counterpart for account), that are needed to be correct with several taxes involved.

manuelcalerosolis committed:
Merge branch '12.0-mig-sale_global_discount' of https://github.com/xtendoo-corporation/sale-workflow into 12.0-mig-sale_global_discount

pedrobaeza commented:
Please don't perform merge commits in the PR, as they screw the commit history for later migrations.

acysos commented on sale_global_discount/models/sale_order.py:
```diff
@@ -0,0 +1,142 @@
+            tax_amount = round_curr(
+                self.get_discounted_global(tax[1], discounts))
+            tax_base = round_curr(
+                self.get_discounted_global(tax[2], discounts)
```
Fail end function:
tax_base = round_curr(
                self.get_discounted_global(tax[2], discounts))

manuelcalerosolis replied:
@acysos Fixed!

Thanks.

manuelcalerosolis committed:
Minor fixed

pedrobaeza commented:
Superseded by #1197
